mythril/analysis/modules/dependence_on_predictable_vars.py

```diff
@@ -39,186 +72,134 @@ def execute(self, state: GlobalState) -> list:
-                            title="Dependence on Predictable Variable",
-                            severity="Low",
-                            description_head="Sending of Ether depends on the blockhash.",
-                            description_tail=description,
-                            swc_id=WEAK_RANDOMNESS,
-                            gas_used=(
-                                call.state.mstate.min_gas_used,
-                                call.state.mstate.max_gas_used,
-                            ),
+                if isinstance(annotation, PredictableValueAnnotation):
+                    state.annotate(
+                        PredictablePathAnnotation(
+                            annotation.operation,
+                            state.get_current_instruction()["address"],
                         )
-                        issues.append(issue)
-                        break
-    return issues
+                    )
 
+        elif opcode == "BLOCKHASH":
 
-def solve(call: Call) -> bool:
-    """
+            param = state.mstate.stack[-1]
 
-    :param call:
-    :return:
-    """
-    try:
-        model = solver.get_model(call.state.mstate.constraints)
-        log.debug("[DEPENDENCE_ON_PREDICTABLE_VARS] MODEL: " + str(model))
-        pretty_model = solver.pretty_print_model(model)
+            try:
+                constraint = [
+                    ULT(param, state.environment.block_number),
+                    ULT(
+                        state.environment.block_number,
+                        symbol_factory.BitVecVal(2 ** 255, 256),
+                    ),
+                ]
+
+                # Why the second constraint? Because without it Z3 returns a solution where param overflows.
+
+                solver.get_model(constraint)
+                state.annotate(OldBlockNumberUsedAnnotation())
+
+            except UnsatError:
+                pass
 
-        log.debug("[DEPENDENCE_ON_PREDICTABLE_VARS] main model: \n%s" % pretty_model)
-        return True
+    else:
+        # we're in post hook
 
-    except UnsatError:
-        log.debug("[DEPENDENCE_ON_PREDICTABLE_VARS] no model found")
-        return False
+        opcode = state.environment.code.instruction_list[state.mstate.pc - 1]["opcode"]
+
+        if opcode == "BLOCKHASH":
+            # if we're in the post hook of a BLOCKHASH op, check if an old block number was used to create it.
+
+            for annotation in state.annotations:
+
+                if isinstance(annotation, OldBlockNumberUsedAnnotation):
+                    state.mstate.stack[-1].annotate(
+                        PredictableValueAnnotation("block hash of a previous block")
+                    )
+        else:
+            # Always create an annotation when COINBASE, GASLIMIT, TIMESTAMP or NUMBER is executed.
+
+            state.mstate.stack[-1].annotate(
+                PredictableValueAnnotation(
+                    "block." + opcode.lower() + " environment variable"
```

Comment:
string formatting might be good here.
```suggestion
                    "block.{} environment variable".format(opcode.lower())
```